functions/create_draft/handler.ts

```diff
@@ -0,0 +1,129 @@
+
+import { CreateDraftFunctionDefinition } from "./mod.ts";
+import { buildDraftBlocks } from "./blocks.ts";
+import {
+  confirmAnnouncementForSend,
+  openDraftEditView,
+  prepareSendAnnouncement,
+  saveDraftEditSubmission,
+} from "./interaction_handler.ts";
+import { ChatPostMessageParams, DraftStatus } from "./types.ts";
+
+import DraftDatastore from "../../datastores/drafts.ts";
+
+/**
+ * This is the handling code for the CreateDraftFunction. It will:
+ * 1. Create a new datastore record with the draft
+ * 2. Build a Block Kit message with the draft and send it to input channel
+ * 3. Update the draft record with the successful sent drafts timestamp
+ * 4. Pause function completion until user interaction
+ */
+export default SlackFunction(
+  CreateDraftFunctionDefinition,
+  async ({ inputs, client }) => {
+    const draftId = crypto.randomUUID();
+
+    // 1. Create a new datastore record with the draft
+    const putResp = await client.apps.datastore.put<
+      typeof DraftDatastore.definition
+    >({
+      datastore: DraftDatastore.name,
+      // @ts-ignore expected fix in future release - otherwise missing non-required items throw type error
+      item: {
+        id: draftId,
+        created_by: inputs.created_by,
+        message: inputs.message,
+        channels: inputs.channels,
+        channel: inputs.channel,
+        icon: inputs.icon,
+        username: inputs.username,
+        status: DraftStatus.Draft,
+      },
+    });
+
+    if (!putResp.ok) {
+      const draftSaveErrorMsg =
+        `Error saving draft announcement. Contact the app maintainers with the following information - (Error detail: ${putResp.error})`;
+      console.log(draftSaveErrorMsg);
+
+      return { error: draftSaveErrorMsg };
+    }
+
+    // 2. Build a Block Kit message with draft announcement and send it to input channel
+    const blocks = buildDraftBlocks(
+      draftId,
+      inputs.created_by,
+      inputs.message,
+      inputs.channels,
+    );
+
+    const params: ChatPostMessageParams = {
+      channel: inputs.channel,
+      blocks: blocks,
+      text: `An announcement draft was posted`,
+    };
+
+    if (inputs.icon) {
+      params.icon_emoji = inputs.icon;
+    }
+
+    if (inputs.username) {
+      params.username = inputs.username;
+    }
+
+    const postDraftResp = await client.chat.postMessage(params);
+    if (!postDraftResp.ok) {
+      const draftPostErrorMsg =
+        `Error posting draft announcement to ${params.channel}. Contact the app maintainers with the following information - (Error detail: ${postDraftResp.error})`;
+      console.log(draftPostErrorMsg);
+
+      return { error: draftPostErrorMsg };
+    }
+
+    // 3. Update the draft record with the successful sent drafts timestamp
+    const putResp2 = await client.apps.datastore.put<
+      typeof DraftDatastore.definition
+    >({
+      datastore: DraftDatastore.name,
+      // @ts-expect-error expecting fix in future SDK release
```

Comment:
What's the type system complaining about here?

Reply:
Missing a datastore item field. 

![Screenshot 2023-01-13 at 9 50 20 AM](https://user-images.githubusercontent.com/55667998/212386748-577d4a6a-22e0-4d25-a717-73ba7d6a3e2d.png)

For example here,`message_ts` is a property that's in the datastore definition but isn't available yet (it's added in a later part of the code).